storybookjs/storybook
# CSF: Automatically extract componentPath

Closes N/A

## What I did

- [x] Automatically extract `rawComponentPath` for index
- [x] Add `componentPath` to index
- [x] Resolve TS config path aliases
- [ ] ~~Automatically add `parameters.componentPath`~~

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

Create a sandbox in this branch and then view its `/index.json`. Verify that `rawComponentPath` is set correctly in the index for various cases (e.g. relative path, absolute path, alias, package import):

Example: https://630511d655df72125520f051-hyyhhtkyeo.chromatic.com/index.json

### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-24396-sha-51edf4aa`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24396-sha-51edf4aa). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-24396-sha-51edf4aa`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24396-sha-51edf4aa) |
| **Triggered by** | @yannbf |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`shilman/auto-component-path`](https://github.com/storybookjs/storybook/tree/shilman/auto-component-path) |
| **Commit** | [`51edf4aa`](https://github.com/storybookjs/storybook/commit/51edf4aaa939c885aa74a748d2489aa856fc6952) |
| **Datetime** | Wed Oct 11 07:16:57 UTC 2023 (`1697008617`) |
| **Workflow run** | [6479231542](https://github.com/storybookjs/storybook/actions/runs/6479231542) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=24396`_
</details>
